openedx/core/djangoapps/waffle_utils/tests/test_init.py

```diff
@@ -7,17 +7,16 @@
 from django.test import TestCase
 from django.test.client import RequestFactory
 from django.test.utils import override_settings
+# Note that we really shouldn't import from edx_toggles' internal API
```

Comment:
Agreed.  Do we have a plan for this?  Does it make sense to have this somehow part of the WaffleFlag api if subclasses need to muck with it?  Other?

Reply:
> Does it make sense to have this somehow part of the WaffleFlag api if subclasses need to muck with it?

No, I don't think so.

> Do we have a plan for this?

The original sin lays in the fact that test_init.py is testing the internal API of the edx_toggles' testutils. The fact that we need to import from `internal` is simply a red flag warning us of this bad pattern. The testing should be moved to edx_toggles. However, those tests are quite complex and I didn't take the time to try to fully understand them. Can we add this as a future action item and do this in a separate PR?

Reply:
Please just add a future task and maybe link back to this.  Thanks.

Reply:
:ok_hand: